### ISSUE TITLE
FFMPEG into Xcode

### ISSUE THREAD

**Issue by clardn** [maculateConception/swift-ffmpeg-player#2]
Hi!

Sorry for getting in touch with you here.

I am trying to reproduce your project, to learn how to use the ffmpeg and integrate it into Xcode.

The problem is that I reproduce your project but I am not able to make the Xcode compile with the ffmpeg files. Could you tell me what tutorial you followed to be able to use ffmpeg as a library in my Xcode project?

Thanks a lot

**Comment by maculateConception** [comment]
Hi @clardn,

Sure, I'll help you get it working. Can you share with me the compilation  errors you're getting ? That'll help me a lot. Also, if you could share your entire project with me  ... perhaps as a zip archive ? Then I will try to run it on my system and  see what's wrong.

So, it was a bit challenging to get it to work initially, but I will try to list a few steps that could help you get it running:

1 - Import the FFmpeg C source directly into Xcode (ffmpeg folder in the project). Note that this project uses  only 4 FFmpeg libraries - avcodec, avformat, avutil, and swresample. Don't import the other ones  like libavfilter, libavdevice, etc.

2 - When Xcode asks  if you  want to create a bridging header (since you  are importing C code into a Swift project), say Yes and it will create Aural-BridgingHeader.h  for you. Inside it, just import ffmpeg.h from this project.

3 - Change some properties in the Xcode target ... such as "Library search path" and "Header search path" and "Framework  search path" and make sure they are correct.

These 3 steps are the main steps to  get your  FFmpeg library code to compile within Xcode.

But, all  these steps have already been performed in the Xcode project in the  repository, so you should  not  need any additional  steps.  So, I would like to see your errors and/or your entire project  to  see what is going wrong for you.

If  you  can share the errors / project, I'll be happy to help you  further.

Cheers :)

**Comment by clardn** [comment]
Hi!

Sorry for the late reply.

It seems like I was frustrated and had tried so many times to get it to work that I was skipping a step.

I rested a bit and started again to follow all the steps you told me and it worked.

In any case, it is not the solution I was looking for, since I need a library that I can use as _ffmpeg.execute ("options")_, as it happens with some libraries that are in git for ios and android.

Anyway it's a start, thank you very much for taking the time to answer me.

**Comment by maculateConception** [comment]
Hi @clardn ,

It's funny you mention that. You want to execute the ffmpeg binary as a child process, correct ? If so, I have a solution for that as well. It is in my other (much bigger) project:

Download the [following zip archive](https://github.com/maculateConception/aural-player/archive/refs/tags/2.6.0.zip), and check the following Swift classes:
- FFmpegWrapper
- CommandExecutor

FFmpegWrapper constructs command line commands to be executed on ffmpeg, and CommandExecutor takes care of spawning the child process and monitoring task progress (such as transcoding).

Basically, it executes commands such as:

**ffmpeg -v quiet -stats -i "File.mp3" -vn -sn "File.m4a"**

Take a look, it might be what you're looking for. If not, oh well.

**Comment by clardn** [comment]
Wow, you are amazing.

I'll be studying swift classes these days, but from what I've seen so far it seems that is what I'm looking for.

Anyway, thank you very much for taking the time to answer me.

All the best
